skyportal/utils/offset_util.py

```diff
@@ -156,26 +313,374 @@ def get_nearby_offset_stars(source_ra, source_dec, source_name,
+                for t, f in cached_files[-max_cache:]:
+                    try:
+                        os.remove(cachedir / f)
+                    except FileNotFoundError:
+                        pass
+    else:
+        hdu = get_hdu(url)
+        if np.count_nonzero(hdu.data) > 0:
+            print("Note: got an empty image")
+            hdu = None
+
+    return hdu
+
+
+def get_finding_chart(source_ra, source_dec, source_name,
+                      image_source='desi',
+                      output_format='pdf',
+                      imsize=3.0,
+                      tick_offset=0.02,
+                      tick_length=0.03,
+                      fallback_image_source='dss',
+                      **offset_star_kwargs):
+
+    """Create a finder chart suitable for spectroscopic observations of
+       the source
+
+    Parameters
+    ----------
+    source_ra : float
+        Right ascension (J2000) of the source
+    source_dec : float
+        Declination (J2000) of the source
+    source_name : str
+        Name of the source
+    image_source : {'desi', 'dss', 'ztfref'}, optional
+        Survey where the image comes from "desi", "dss", "ztfref" (more to be added)
+    output_format : str, optional
+        "pdf" of "png" -- determines the format of the returned finder
+    imsize : float, optional
+        Requested image size (on a size) in arcmin. Should be between 2-15.
+    tick_offset : float, optional
+        How far off the each source should the tick mark be made? (in arcsec)
+    tick_length : float, optional
+        How long should the tick mark be made? (in arcsec)
+    fallback_image_source : str, optional
+        Where what `image_source` should we fall back to if the
+        one requested fails
+    **offset_star_kwargs : dict, optional
+        Other parameters passed to `get_nearby_offset_stars`
+
+    Returns
+    -------
+    dict
+        name : str
+            suggested filename based on `source_name` and `output_format`
+        data : str
+            binary encoded data for the image (to be streamed)
+    """
+    if (imsize < 2.0) or (imsize > 15):
+        raise ValueError("Requested `imsize` out of range")
+
+    if image_source not in source_image_parameters:
+            return {'success': False, 'reason': 'image source not in list'}
+
+    fig = plt.figure(figsize=(11, 8.5), constrained_layout=False)
+    widths = [2.6, 1]
+    heights = [2.6, 1]
+    spec = fig.add_gridspec(ncols=2, nrows=2, width_ratios=widths,
+                            height_ratios=heights, left=0.05, right=0.95)
+
+    npixels = source_image_parameters[image_source].get("npixels", 256)
+    pixscale = 60*imsize/npixels
+
+    hdu = fits_image(source_ra, source_dec, imsize=imsize,
+                     image_source=image_source)
+
+    # skeleton WCS - this is the field that the user requested
+    # North up, East left
+    wcs = WCS(naxis=2)
+    wcs.wcs.crpix = [npixels/2, npixels/2]
```

Comment:
What is the logic here?  There are lots of calculations in this file with magic numbers, such as `60*imsize/npixels` above.  It would be good to have motivations for how those numbers were chosen and what they mean in comments.

Reply:
Done.